src/storj.c

```diff
@@ -242,6 +242,77 @@ static void get_bucket_request_worker(uv_work_t *work)
     }
 }
 
+static void get_bucket_id_request_worker(uv_work_t *work)
+{
+    get_bucket_id_request_t *req = work->data;
+    int status_code = 0;
+
+    // Derive a key based on the master seed and bucket name magic number
+    char *bucket_key_as_str = calloc(DETERMINISTIC_KEY_SIZE + 1, sizeof(char));
+    generate_bucket_key(req->encrypt_options->mnemonic,
+                        BUCKET_NAME_MAGIC,
+                        &bucket_key_as_str);
+
+    uint8_t *bucket_key = str2hex(strlen(bucket_key_as_str), bucket_key_as_str);
+    if (!bucket_key) {
+        req->error_code = STORJ_MEMORY_ERROR;
+        return;
+    }
+
+    free(bucket_key_as_str);
```

Comment:
this variable never gets freed when returning on line 259. handle frees and return using goto at end of function

Reply:
OK. I see it now. However, the same issues exists on many other places in the code.
I would suggest to move the code for generating the bucket key in a separate function that handles the memory correctly and replace it all over. Rather than polluting the code with `goto`s. Sounds good?

Reply:
we've been using gotos across the entire library. You could just put variables you need access to outside on the heap and the rest on the stack. I don't exactly understand what you are saying. maybe show me an example of what you mean

Reply:
Examples of exactly the same code with the same issue:

https://github.com/Storj/libstorj/blob/master/src/storj.c#L25-L37

https://github.com/Storj/libstorj/blob/master/src/storj.c#L111-L123

https://github.com/Storj/libstorj/blob/master/src/storj.c#L189-L201

https://github.com/Storj/libstorj/blob/master/src/storj.c#L267-L279

https://github.com/Storj/libstorj/blob/master/src/storj.c#L357-L369

https://github.com/Storj/libstorj/blob/e46345f1b80291729d50345d746dabb6cb8d039c/src/uploader.c#L2473-L2485